ringpop.go

```diff
@@ -352,6 +359,31 @@ func (rp *Ringpop) setState(s state) {
 	}
 }
 
+// RegisterSelfEvictHook registers the eviction hooks that need to be executed
+// before and after self eviction from the membership. An error will be returned
+// if ringpop was unable to register the hooks. This could happen in the following
+// cases:
+// - Ringpop has not been bootstrapped yet
+// - SelfEvict has already been called
+// - The hook is already registered
+func (rp *Ringpop) RegisterSelfEvictHook(hooks swim.SelfEvictHook) error {
+	if !rp.Ready() {
```

Comment:
What is the reason that we prevent self-evict hooks from being registered before bootstrapping?

From an API perspective, I can imagine in an application lifecycle that I would want to have a setup/init where I register all my hooks and then set the world running by calling `Bootstrap()` (much the same way you'd register handlers before calling `Listen()`).